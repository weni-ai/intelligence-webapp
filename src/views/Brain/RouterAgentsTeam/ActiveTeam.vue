<template>
  <section class="active-team">
    <UnnnicIntelligenceText
      tag="h2"
      family="secondary"
      size="body-lg"
      color="neutral-darkest"
      weight="bold"
      data-testid="title"
    >
      {{ $t('router.agents_team.active_team.title') }}
    </UnnnicIntelligenceText>

    <section
      v-if="isLoadingTeam || activeTeam.length"
      class="active-team__cards"
    >
      <template v-if="isLoadingTeam">
        <AssignAgentCard
          v-for="(_, index) in Array(3)"
          :key="index"
          loading
          data-testid="loading-card"
        />
      </template>

      <template v-else>
        <AssignAgentCard
          v-for="agent in activeTeam"
          :key="agent.uuid"
          :agent="agent"
          :assignment="false"
          data-testid="team-card"
        />
      </template>
    </section>

    <section
      v-if="!isLoadingTeam && activeTeam.length === 0"
      class="active-team__empty"
      data-testid="empty-state"
    >
      <UnnnicIntelligenceText
        tag="p"
        family="secondary"
        size="body-gt"
        color="neutral-dark"
        weight="bold"
      >
        {{ $t('router.agents_team.active_team.no_team') }}
      </UnnnicIntelligenceText>

      <section class="empty__description">
        <!-- eslint-disable-next-line vue/component-name-in-template-casing -->
        <i18n-t
          class="description__text"
          keypath="router.agents_team.active_team.no_team_description"
          tag="p"
        >
          <template #link>
            <a
              class="description__link"
              :href="agentsTeamStore.linkToCreateAgent"
              target="_blank"
              >{{ $t('router.agents_team.active_team.create_agent') }}</a
            >
          </template>
        </i18n-t>
      </section>
    </section>
  </section>
</template>

<script setup>
import { computed, onMounted } from 'vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

import AssignAgentCard from '@/components/Brain/AgentsTeam/AssignAgentCard.vue';

const agentsTeamStore = useAgentsTeamStore();
const activeTeam = computed(
  () => agentsTeamStore.activeTeam.data?.agents || [],
);

const isLoadingTeam = computed(
  () => agentsTeamStore.activeTeam.status === 'loading',
);

onMounted(() => {
  agentsTeamStore.loadActiveTeam();
});
</script>

<style lang="scss" scoped>
.active-team {
  border-radius: $unnnic-border-radius-md;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

  padding: $unnnic-spacing-sm;

  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-sm;

  &__empty {
    border-radius: $unnnic-border-radius-md;
    border: $unnnic-border-width-thinner dashed $unnnic-color-neutral-cleanest;

    padding: $unnnic-spacing-xl $unnnic-spacing-sm;

    display: flex;
    flex-direction: column;
    align-items: center;

    background-color: $unnnic-color-neutral-lightest;

    .empty__description {
      display: flex;
      align-items: center;
      gap: 2px;

      .description__text {
        font-family: $unnnic-font-family-secondary;
        font-size: $unnnic-font-size-body-md;
        color: $unnnic-color-neutral-cloudy;
      }

      .description__link {
        font-size: $unnnic-font-size-body-md;
        color: $unnnic-color-neutral-cloudy;
        text-decoration: underline;
      }
    }
  }

  &__cards {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: $unnnic-spacing-sm;
  }
}
</style>
