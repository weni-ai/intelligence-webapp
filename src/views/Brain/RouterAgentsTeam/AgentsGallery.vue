<template>
  <section
    class="agents-gallery"
    data-testid="agents-gallery"
  >
    <UnnnicIntelligenceText
      tag="h2"
      family="secondary"
      size="body-lg"
      color="neutral-darkest"
      weight="bold"
    >
      {{ $t('router.agents_team.gallery.title') }}
    </UnnnicIntelligenceText>

    <UnnnicTab
      class="agents-gallery__tabs"
      :tabs="contentTabs.map((tab) => tab.page)"
      :activeTab="activeTab"
      data-testid="agents-gallery-tabs"
      @change="onTabChange"
    >
      <template
        v-for="tab in contentTabs"
        :key="tab.page"
        #[`tab-head-${tab.page}`]
      >
        {{ $t(`router.agents_team.gallery.${tab.title}`) }}
      </template>
    </UnnnicTab>

    <UnnnicInput
      v-model="search[activeTab]"
      iconLeft="search"
      :placeholder="$t('router.agents_team.gallery.search_placeholder')"
      data-testid="search-input"
    />

    <section
      :class="[
        'agents-gallery__cards',
        {
          'agents-gallery__cards--empty': isMyAgentsEmpty,
        },
      ]"
      data-testid="agent-cards"
    >
      <template v-if="isLoadingAgents">
        <AgentCard
          v-for="(_, index) in Array(3)"
          :key="index"
          loading
          data-testid="agent-card-loading"
        />
      </template>

      <template v-else>
        <UnnnicIntelligenceText
          v-if="isSearchEmpty"
          color="neutral-clean"
          family="secondary"
          size="body-gt"
          tag="p"
        >
          {{ $t('router.agents_team.gallery.no_agent_found') }}
        </UnnnicIntelligenceText>

        <template v-else>
          <AgentCard
            v-if="activeTab === 'my-agents'"
            data-testid="agent-card-empty"
            empty
          />

          <AgentCard
            v-for="agent in agentsData"
            :key="agent.uuid"
            :title="agent.name"
            :description="agent.description"
            :skills="agent.skills"
            :uuid="agent.uuid"
            :assigned="agent.assigned"
            data-testid="agent-card"
          />
        </template>
      </template>
    </section>
  </section>
</template>

<script setup>
import { computed, onMounted, ref, watch } from 'vue';
import { debounce } from 'lodash';

import AgentCard from '@/components/Brain/AgentsTeam/AgentCard.vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

const agentsTeamStore = useAgentsTeamStore();
const officialAgents = agentsTeamStore.officialAgents;
const myAgents = agentsTeamStore.myAgents;

const contentTabs = ref([
  { title: 'official', page: 'official' },
  { title: 'my-agents', page: 'my-agents' },
]);
const activeTab = ref(contentTabs.value[0].page);
const search = ref({
  official: '',
  'my-agents': '',
});
const isSearchEmpty = ref(false);

const agentsData = computed(() =>
  activeTab.value === 'official' ? officialAgents.data : myAgents.data,
);
const isLoadingAgents = computed(
  () => officialAgents.status === 'loading' || myAgents.status === 'loading',
);
const isMyAgentsEmpty = computed(
  () =>
    activeTab.value === 'my-agents' &&
    myAgents.data.length === 0 &&
    !isLoadingAgents.value,
);

function updateSearchEmpty() {
  const currentAgents = agentsData.value;
  const currentSearch = search.value[activeTab.value];

  isSearchEmpty.value =
    currentSearch.trim() !== '' && currentAgents.length === 0;
}

function onTabChange(newTab) {
  activeTab.value = newTab;

  if (agentsTeamStore.myAgents.status === null && newTab === 'my-agents') {
    agentsTeamStore.loadMyAgents();
  }

  updateSearchEmpty();
}

onMounted(() => {
  agentsTeamStore.loadOfficialAgents();
});

const debouncedSearch = (callback) => debounce(callback, 300);
watch(
  () => search.value['my-agents'],
  debouncedSearch(async (search) => {
    await agentsTeamStore.loadMyAgents({ search });
    updateSearchEmpty();
  }),
);
watch(
  () => search.value['official'],
  debouncedSearch(async (search) => {
    await agentsTeamStore.loadOfficialAgents({ search });
    updateSearchEmpty();
  }),
);
</script>

<style lang="scss" scoped>
.agents-gallery {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-sm;

  &__tabs {
    margin-bottom: -$unnnic-spacing-sm;
  }

  &__cards {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: $unnnic-spacing-sm;

    &--empty {
      grid-template-columns: repeat(1, 1fr);
    }
  }
}
</style>
