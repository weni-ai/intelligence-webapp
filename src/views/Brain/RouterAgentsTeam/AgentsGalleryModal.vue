<template>
  <UnnnicModalDialog
    class="agents-gallery-modal"
    :modelValue="agentsTeamStore.isAgentsGalleryOpen"
    showCloseIcon
    showActionsDivider
    :title="$t('router.agents_team.gallery.title')"
    size="lg"
    @update:model-value="closeModal"
  >
    <section
      class="agents-gallery"
      data-testid="agents-gallery"
    >
      <SidebarMenu
        class="agents-gallery__sidebar"
        data-testid="sidebar-menu"
      >
        <SideBarItem
          v-for="tab in contentTabs"
          :key="tab.page"
          class="sidebar__item"
          :text="$t(`router.agents_team.gallery.${tab.title}`)"
          :active="tab.page === activeTab"
          :iconFilled="tab.page === activeTab"
          data-testid="sidebar-item"
          @click="onTabChange(tab.page)"
        />
      </SidebarMenu>

      <section class="agents-gallery__content">
        <UnnnicInput
          v-if="!isMyAgentsEmpty"
          v-model="search[activeTab]"
          iconLeft="search"
          :placeholder="$t('router.agents_team.gallery.search_placeholder')"
          data-testid="search-input"
        />

        <section
          v-if="activeTab === 'my-agents'"
          :class="[
            'agents-gallery__custom-agents',
            {
              'agents-gallery__custom-agents--empty': isMyAgentsEmpty,
            },
          ]"
        >
          <UnnnicIntelligenceText
            tag="p"
            family="secondary"
            size="body-gt"
            color="neutral-cloudy"
          >
            {{ $t('router.agents_team.gallery.custom_agents_description') }}
          </UnnnicIntelligenceText>

          <!-- This comment prevents from auto-capitalizing i18n-t to I18nT which would break the component -->
          <!-- eslint-disable-next-line vue/component-name-in-template-casing -->
          <i18n-t
            class="custom-agents__description"
            keypath="router.agents_team.gallery.weni_cli_documentation_description"
            tag="p"
          >
            <template #weni_cli_documentation>
              <p
                class="description__link"
                @click="openCLI"
              >
                {{ $t('router.agents_team.gallery.weni_cli_documentation') }}
              </p>
            </template>
          </i18n-t>
        </section>

        <template v-if="!isMyAgentsEmpty">
          <section
            :class="[
              'content__cards',
              {
                'content__cards--empty': isMyAgentsEmpty,
              },
            ]"
            data-testid="agent-cards"
          >
            <template v-if="isLoadingAgents">
              <AssignAgentCard
                v-for="(_, index) in Array(3)"
                :key="index"
                loading
                data-testid="agent-card-loading"
              />
            </template>

            <template v-else>
              <UnnnicIntelligenceText
                v-if="isSearchEmpty"
                color="neutral-clean"
                family="secondary"
                size="body-gt"
                tag="p"
              >
                {{ $t('router.agents_team.gallery.no_agent_found') }}
              </UnnnicIntelligenceText>

              <template v-else>
                <AssignAgentCard
                  v-for="agent in agentsData"
                  :key="agent.uuid"
                  :agent="agent"
                  @agent-assigned="closeModal"
                />
              </template>
            </template>
          </section>
        </template>
      </section>
    </section>
  </UnnnicModalDialog>
</template>

<script setup>
import { computed, onMounted, ref, watch } from 'vue';
import { debounce } from 'lodash';

import AssignAgentCard from '@/components/Brain/AgentsTeam/AssignAgentCard.vue';
import SideBarItem from '@/components/Sidebar/SideBarItem.vue';
import SidebarMenu from '@/components/Sidebar/SidebarMenu.vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

const agentsTeamStore = useAgentsTeamStore();
const officialAgents = agentsTeamStore.officialAgents;
const myAgents = agentsTeamStore.myAgents;

const contentTabs = ref([
  { title: 'official_agents', page: 'official' },
  { title: 'custom_agents', page: 'my-agents' },
]);
const activeTab = ref(contentTabs.value[0].page);
const search = ref({
  official: '',
  'my-agents': '',
});
const isSearchEmpty = ref(false);

const agentsData = computed(() =>
  activeTab.value === 'official' ? officialAgents.data : myAgents.data,
);
const isLoadingAgents = computed(
  () => officialAgents.status === 'loading' || myAgents.status === 'loading',
);
const isMyAgentsEmpty = computed(
  () =>
    activeTab.value === 'my-agents' &&
    myAgents.data.length === 0 &&
    !isSearchEmpty.value &&
    !isLoadingAgents.value,
);

function updateSearchEmpty() {
  const currentAgents = agentsData.value;
  const currentSearch = search.value[activeTab.value];

  isSearchEmpty.value =
    currentSearch.trim() !== '' && currentAgents.length === 0;
}

function onTabChange(newTab) {
  activeTab.value = newTab;

  if (agentsTeamStore.myAgents.status === null && newTab === 'my-agents') {
    agentsTeamStore.loadMyAgents();
  }

  updateSearchEmpty();
}

function closeModal() {
  agentsTeamStore.isAgentsGalleryOpen = false;
}

function openCLI() {
  window.open(agentsTeamStore.linkToCreateAgent, '_blank');
}

onMounted(() => {
  agentsTeamStore.loadOfficialAgents();
});

const debouncedSearch = (callback) => debounce(callback, 300);
watch(
  () => search.value['my-agents'],
  debouncedSearch(async (search) => {
    await agentsTeamStore.loadMyAgents({ search });
    updateSearchEmpty();
  }),
);
watch(
  () => search.value['official'],
  debouncedSearch(async (search) => {
    await agentsTeamStore.loadOfficialAgents({ search });
    updateSearchEmpty();
  }),
);
</script>

<style lang="scss" scoped>
.agents-gallery-modal {
  :deep(.unnnic-modal-dialog__container) {
    width: 75vw;
    height: 85vh;
  }

  .agents-gallery {
    height: 100%;

    display: grid;
    grid-template-columns: 170px 1fr;
    gap: $unnnic-spacing-sm;

    &__sidebar {
      .sidebar__item {
        padding: $unnnic-spacing-ant;
      }
    }

    &__content {
      display: flex;
      flex-direction: column;
      gap: $unnnic-spacing-sm;

      .content__cards {
        margin-bottom: $unnnic-spacing-sm;

        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
        gap: $unnnic-spacing-sm;

        &--empty {
          grid-template-columns: repeat(1, 1fr);
        }
      }
    }

    &__custom-agents {
      height: fit-content;

      display: flex;
      align-items: center;
      justify-content: flex-start;
      gap: $unnnic-spacing-nano;

      &--empty {
        flex-direction: column;
        justify-content: center;

        width: 100%;
        height: 100%;
      }

      .custom-agents__description {
        display: flex;
        align-items: center;
        gap: $unnnic-spacing-nano;

        font-size: $unnnic-font-size-body-gt;

        .description__link {
          position: relative;

          font-size: $unnnic-font-size-body-gt;
          color: $unnnic-color-neutral-cloudy;
          font-weight: $unnnic-font-weight-bold;

          cursor: pointer;

          &::after {
            content: '';

            position: absolute;
            bottom: 0;
            left: 0;

            width: 100%;
            height: 1px;

            background-color: $unnnic-color-neutral-cloudy;
          }
        }
      }
    }
  }
}
</style>
