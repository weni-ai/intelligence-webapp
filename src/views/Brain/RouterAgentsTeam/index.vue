<template>
  <section class="agents-team">
    <UnnnicDivider
      ySpacing="lg"
      class="agents-team__divider"
    />

    <ActiveTeam />
  </section>
</template>

<script setup>
import ActiveTeam from './ActiveTeam.vue';
</script>

<style lang="scss" scoped>
section.agents-team {
  display: grid;
  grid-template-rows: auto 1fr;

  .agents-team__divider {
    margin-top: 0;
  }
}
</style>
