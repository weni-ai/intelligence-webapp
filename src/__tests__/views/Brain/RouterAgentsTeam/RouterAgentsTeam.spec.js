import { shallowMount } from '@vue/test-utils';

import RouterAgentsTeam from '@/views/Brain/RouterAgentsTeam/index.vue';
import ActiveTeam from '@/views/Brain/RouterAgentsTeam/ActiveTeam.vue';
import AgentsGallery from '@/views/Brain/RouterAgentsTeam/AgentsGallery.vue';
import UnnnicDivider from '@/components/Divider.vue';

describe('RouterAgentsTeam', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(RouterAgentsTeam);
  });

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders the all components', () => {
    expect(wrapper.findComponent(ActiveTeam).exists()).toBe(true);
    expect(wrapper.findComponent(UnnnicDivider).exists()).toBe(true);
    expect(wrapper.findComponent(AgentsGallery).exists()).toBe(true);
  });
});
