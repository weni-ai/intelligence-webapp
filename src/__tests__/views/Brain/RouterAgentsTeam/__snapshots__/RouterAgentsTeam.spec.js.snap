// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RouterAgentsTeam > matches snapshot 1`] = `
<section
  class="agents-team"
  data-v-d0fbe27f=""
>
  <unnnic-divider-stub
    class="agents-team__divider"
    data-v-d0fbe27f=""
    scheme="neutral-soft"
    yspacing="lg"
  />
  <active-team-stub
    data-v-d0fbe27f=""
  />
</section>
`;
